### PR TITLE
[expo-contacts][android] Implement contact group functions for android

### PR DESCRIPTION
**Pull request by devdave25**
# Why

Currently expo-contacts only supports ios groups. Looking to have groups when using expo-contacts on android.
This PR adds android support for the following functions:
- getGroupsAsync(groupQuery)
- createGroupAsync(name)
- updateGroupName(groupName, groupId)
- removeGroupAsync(groupId)
- addExistingContactToGroupAsync(contactId, groupId)
- removeContactFromGroupAsync(contactId, groupId)

Additionally implemented fetching contacts by groupId using `getContactsAsync({ groupId })`

# How

Build these changes utilising the existing ios functions and based the changes on the existing android contact functions.

# Test Plan

Please forgive me for I have been unable to build expo locally and so these changes haven't fully been tested. This is my first contibution to this project. I would need some support with preparing the documentation and performing tests on this PR.

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.kt [hunk 1]
```diff
@@ -223,6 +256,103 @@ class ContactsModule : Module() {
...
+      val uri: Uri = resolver.insert(ContactsContract.Groups.CONTENT_URI, contentValues)
+      val groupId = uri.lastPathSegment?.toLongOrNull()
+    }
+
+    AsyncFunction("updateGroupName") { groupName: String?, groupId: String? ->
+      ensurePermissions()
+
+      var targetGroup = getGroupById(groupId)
+
+      if (targetGroup != null) {
```
Comment (lukmccall) [inline packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.kt:300]:
```suggestion
      if (getGroupById(groupId) != null) {
```

File: packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.kt [hunk 2]
```diff
@@ -223,6 +256,103 @@ class ContactsModule : Module() {
...
+      val rowsDeleted = resolver.delete(uri, null, null)
+      return rowsDeleted
+    }
+
+    AsyncFunction("addExistingContactToGroupAsync") { contactId: String?, groupId: String? ->
+      ensurePermissions()
+
+      var targetGroup = getGroupById(groupId)
+
+      if (targetGroup != null) {
```
Comment (lukmccall) [inline packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.kt:326]:
```suggestion
      if (getGroupById(groupId) != null) {
```

File: packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.kt [hunk 3]
```diff
@@ -223,6 +256,103 @@ class ContactsModule : Module() {
...
+        throw GroupNotFoundException()
+      }
+    }
+
+    AsyncFunction("removeContactFromGroupAsync") { contactId: String?, groupId: String? ->
+      ensurePermissions()
+
+      var targetGroup = getGroupById(groupId)
+
+      if (targetGroup != null) {
```
Comment (lukmccall) [inline packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.kt:344]:
```suggestion
      if (getGroupById(groupId) != null) {
```

File: packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.kt [hunk 4]
```diff
@@ -644,6 +810,39 @@ class ContactsModule : Module() {
...
+      cr.query(
+        ContactsContract.Groups.CONTENT_URI,
+        arrayOf(ContactsContract.Groups._ID, ContactsContract.Groups.TITLE),
+        null,
+        null,
+        null
+      )
+    }?.use { cursor ->
+      groups = loadGroupsFrom(cursor)
+
```
Comment (lukmccall) [inline packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.kt:840]:
```suggestion
```

**Review by lukmccall — COMMENTED** [review]
So far, everything looks good. Could you also add the TypeScript part for the new methods?

File: packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.kt (hunk 4 above)
Comment (devdave25) [reply packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.kt:830]: Thank you for your review on this. I have updated the typescript documentation now.
This is the only change I am unsure about exactly what you are asking me to do, thanks.

**Review by lukmccall — COMMENTED** [review]
"Looks good. Thanks for updating the TS file. There are two more things that need to be done before merging:
1. Could you rebuild the JS files by running `yarn build` in the `expo-contacts` directory?
2. Could you also add the changelog entry?

**Comment by devdave25** [comment]
I have updated the changelogs and completed the rebuild of js. Thanks.

**Review by lukmccall — APPROVED** [review]
LGTM ✅ 
I'll merge it as soon as the CI pass!

**Comment by lukmccall** [comment]
Unfortunately, it doesn't compile:
```
e: file:///home/runner/work/expo/expo/packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.kt:285:39 Unresolved reference: context

e: file:///home/runner/work/expo/expo/packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.kt:286:27 Unresolved reference: ContentValues
> Task :expo-contacts:compileDebugKotlin FAILED
e: file:///home/runner/work/expo/expo/packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.kt:287:9 Unresolved reference: put
e: file:///home/runner/work/expo/expo/packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.kt:290:22 Type mismatch: inferred type is Uri? but Uri was expected
e: file:///home/runner/work/expo/expo/packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.kt:298:41 Unresolved reference: context
e: file:///home/runner/work/expo/expo/packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.kt:299:29 Unresolved reference: ContentValues
e: file:///home/runner/work/expo/expo/packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.kt:300:11 Unresolved reference: put
e: file:///home/runner/work/expo/expo/packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.kt:315:7 'return' is not allowed here
e: file:///home/runner/work/expo/expo/packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.kt:315:14 Type mismatch: inferred type is ModuleDefinitionData but Int was expected
e: file:///home/runner/work/expo/expo/packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.kt:322:29 Unresolved reference: ContentValues
e: file:///home/runner/work/expo/expo/packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.kt:323:11 Unresolved reference: put
e: file:///home/runner/work/expo/expo/packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.kt:324:11 Unresolved reference: put
e: file:///home/runner/work/expo/expo/packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.kt:325:11 Unresolved reference: put
e: file:///home/runner/work/expo/expo/packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.kt:334:5 Cannot use 'Nothing' as reified type parameter
e: file:///home/runner/work/expo/expo/packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.kt:343:9 'return' is not allowed here
e: file:///home/runner/work/expo/expo/packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.kt:343:16 Type mismatch: inferred type is ModuleDefinitionData but Int was expected
```

**Pull request closed without merging**